src/language/formatter/abstractFormatter.ts

```diff
@@ -20,7 +20,7 @@ import { IFormatter, IFormatterMetadata } from "./formatter";
 
 export abstract class AbstractFormatter implements IFormatter {
     public static metadata: IFormatterMetadata;
-    public abstract format(failures: RuleFailure[]): string;
+    public abstract format(failures: RuleFailure[], fixes?: RuleFailure[], fileNames?: string[]): string;
```

Comment:
correct me if i'm wrong: these parameters in the interface definition should not have `?` as they are always passed to the implementation. implementors should instead choose to _omit_ the args from their implementations if unused.

Reply:
@giladgray I think you're right. I was following the convention already on `fixes` in `IFormatter`, however, since the args are always provided they shouldn't be marked as optional.
I've pushed a patch to remove the `?`s.